### ISSUE TITLE
Shift_JIS misdetected

### ISSUE THREAD

**Issue by scuti-dai** [amake/flutter_charset_detector#5]
Uint8List: [121, 45, 116, 111, 103, 105, 121, 97, 47, 42, 47, 251, 121, 146, 74, 32, 151, 83, 137, 238, 47, 42, 47]

result.charset: WINDOWS-1252
result.string: y-togiya/*/ûy’J —S‰î/*/

it should be is 'SHIFT_JIS' and y-togiya/*/硎谷 祐介/*/

**Comment by amake** [comment]
The underlying detection engines are different on iOS and Android, so it would be helpful to know which platform you're using, but regardless:

If your entire input is just the byte list you specified, then it's probably too short. Only some encodings can be conclusively identified from an arbitrary-length snippet; for others detection is probabilistic: given enough text, the engine can guess the probability of some result.

WINDOWS-1252 is probably a fallback for when a specific other encoding wasn't detected.

**Comment by scuti-dai** [comment]
@amake That is all byte my BE response. 
flutter: "3.10.6"

**Comment by amake** [comment]
@scuti-dai I'm asking you if you're using iOS or Android, because the *detection result* can be different depending on the platform.

**Comment by scuti-dai** [comment]
@amake sorry, i am using ios

**Comment by amake** [comment]
OK, that's based on [uchardet](https://www.freedesktop.org/wiki/Software/uchardet/).

I tried decoding your bytes in Python:

```python
b = bytes([121, 45, 116, 111, 103, 105, 121, 97, 47, 42, 47, 251, 121, 146, 74, 32, 151, 83, 137, 238, 47, 42, 47])
print(b) => b'y-togiya/*/\xfby\x92J \x97S\x89\xee/*/'
print(b.decode('shift_jis')) => UnicodeDecodeError: 'shift_jis' codec can't decode byte 0xfb in position 11: illegal multibyte sequence
b.decode('shift_jis_2004') => 'y-togiya/*/𨵱谷 祐介/*/'
```

Your bytes seem to encode a character that is not in Shift JIS, but is in Shift JIS 2004, which I guess uchardet doesn't support.

I tried saving a Shift JIS file with contents `y-togiya/*/谷 祐介/*/` and it seems it's too short for uchardet to recognize, as it still gave the result WINDOWS-1252.

So there's nothing I can do for you. Your snippet is not strictly valid Shift JIS to begin with (it's Shift JIS 2004 or other extension), and even if it was it's too short for the underlying engine to detect.
